src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -464,10 +466,10 @@ public void EnqueueRange(IEnumerable<TElement> elements, TPriority priority)
             ArgumentNullException.ThrowIfNull(elements);
 
             int count;
-            if (elements is ICollection<(TElement Element, TPriority Priority)> collection &&
```

Comment:
This was previously checking for the wrong ICollection type resulting in the optimization never being used.